src/core/connection.c

```diff
@@ -5451,6 +5451,13 @@ QuicConnRecvDatagrams(
             goto Drop;
         }
 
+#ifdef QUIC_USE_RAW_DATAPATH
+        if (DatagramPath->Route.State == RouteResolved && !DatagramPath->Route.QueueUpdated) {
+            DatagramPath->Route.Queue = Datagram->Route->Queue;
```

Comment:
Seems weird that elsewhere we use a helper function CxPlatDpRawAssignQueue (with different XDP and DPDK implementations) to abstract the assignment of the queue, but then here we just dig in and set it directly. Does this code break dpdk?

Reply:
Also, what is going on here anyway? We already assign a queue in CxPlatResolveRoute, but here we are overriding that with the packet's assigned queue, is that right? I am guessing this is done to automatically mirror the RSS configuration or something like that, but then my question is why do we assign a queue in CxPlatResolveRoute?

Also, I don't see anything in the PR description about this change.

Reply:
DPDK has been defunct (can't compile and won't work) for a long time.

Reply:
I think my second question is the more important one here- sorry for putting it into a thread instead of making a new comment.

Reply:
>> We already assign a queue in CxPlatResolveRoute, but here we are overriding that with the packet's assigned queue, is that right? I am guessing this is done to automatically mirror the RSS configuration or something like that, but then my question is why do we assign a queue in CxPlatResolveRoute?

The workflow here is we resolve the route and assign a default queue. When we receive the first packet after route resolution, we override the queue to match RSS. If we don't give it a default queue in CxPlatResolveRoute, sends won't go through.

Reply:
Got it, that makes sense and I think it justifies the addition of the QueueUpdated flag to CXPLAT_ROUTE. Please add a code comment to the QueueUpdated flag to explain this behavior of choosing a "default" queue and then updating the queue based on the received packet.

But hey wait, why don't we just always compare the queue of inbound traffic to the current queue and update to match it if it's different? That way we would stay up to date with an RSS configuration change.

Reply:
That seems better. I will update the PR.

Reply:
LGTM